backend/test/watches.spec.js

```diff
@@ -0,0 +1,527 @@
+      ])
+
+      fooBarRoom = new Room('foo', [
+        { type: 'DELETED', object: foobar }
+      ])
+
+      fooBazRoom = new Room('foo', [
+        { type: 'DELETED', object: foobaz }
+      ])
+
+      fooIssuesRoom = new Room('foo', [])
+
+      emitter.emit('event', { type: 'DELETED', object: foobar })
+      emitter.emit('event', { type: 'DELETED', object: foobaz })
+
+      expect(errorSpy).to.be.calledOnce
+      expect(isResourcePendingStub).to.be.calledTwice
+      expect(removePendingResourceStub).to.be.calledOnce
+      expect(deleteJournalsStub).to.be.calledTwice
+
+      expect(fooRoom.events).to.be.empty
+      expect(fooBarRoom.events).to.be.empty
+      expect(fooBazRoom.events).to.be.empty
+      expect(fooIssuesRoom.events).to.be.empty
+    })
+  })
+
+  describe('journals', function () {
+    const serviceUnavailable = new Error('Service Unavailable')
+    serviceUnavailable.status = 503
+
+    const issueEvent = {}
+    const issuesRoom = {
+      emit (...args) {
+        expect(args).to.eql(['events', {
+          kind: 'issues',
+          events: [issueEvent]
+        }])
+      }
+    }
+
+    const commentEvent = {
+      object: {
+        metadata: {
+          namespace: 'foo',
+          name: 'bar'
+        }
+      }
+    }
+    const commentsRoom = {
+      emit (...args) {
+        expect(args).to.eql(['events', {
+          kind: 'comments',
+          events: [commentEvent]
+        }])
+      }
+    }
+
+    const nsp = {
+      to (room) {
+        switch (room) {
+          case 'issues':
+            return issuesRoom
+          case 'comments_foo/bar':
+            return commentsRoom
+        }
+      }
+    }
+
+    const io = {
+      of (namespace) {
+        expect(namespace).to.equal('/journals')
+        return nsp
+      }
+    }
+
+    const journalCache = {
+      onIssue (handler) {
+        handler(issueEvent)
+      },
+      onComment (handler) {
+        handler(commentEvent)
+      }
+    }
+    let gitHubConfigStub
+    let warnSpy
+    let infoSpy
+    let errorSpy
+    let cacheStub
+    let loadOpenIssuesStub
+
+    beforeEach(function () {
+      gitHubConfigStub = sandbox.stub(config, 'gitHub')
+      infoSpy = sandbox.spy(logger, 'info')
+      warnSpy = sandbox.spy(logger, 'warn')
+      errorSpy = sandbox.spy(logger, 'error')
+      cacheStub = sandbox.stub(cache, 'getJournalCache')
+      loadOpenIssuesStub = sandbox.stub(journals, 'loadOpenIssues')
+    })
+
+    it('should log missing gitHub config', async function () {
+      gitHubConfigStub.get(() => false)
+      watches.journals(io)
+      expect(warnSpy).to.be.calledOnce
+    })
+
+    it('should watch journals', async function () {
+      gitHubConfigStub.get(() => true)
+      cacheStub.returns(journalCache)
+      loadOpenIssuesStub.onCall(0).throws(serviceUnavailable)
+      watches.journals(io, { minTimeout: 1 })
+      expect(cacheStub).to.be.calledOnce
+      await delay(5)
```

Comment:
use explicit finish event instead of fixed delay

Reply:
✔️